### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by mfnsvrtm**

### CODE REVIEW

File: task2-sql-db-design/script.sql [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    bank_account_number VARCHAR(30) NOT NULL
+);
+
+CREATE TABLE "user" (
+    id SERIAL PRIMARY KEY,
+    username VARCHAR(30) NOT NULL,
+    password VARCHAR(30) NOT NULL,
+    first_name VARCHAR(30) NOT NULL,
+    last_name VARCHAR(30) NOT NULL,
+    email VARCHAR(30) NOT NULL,
```
Comment (fedorloktev) [inline task2-sql-db-design/script.sql:41]: I am not sure 30 symbols limit is enough for e-mail format

File: task2-sql-db-design/script.sql [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    timer TIMESTAMPTZ NOT NULL
+);
+
+CREATE TABLE purchase (
+    id SERIAL PRIMARY KEY,
+    item_id SERIAL NOT NULL,
+    bid_id SERIAL NOT NULL,
+    amount DECIMAL(9, 2) NOT NULL,
+    payment_method VARCHAR(30) NOT NULL,
+    bank_account_number VARCHAR(30) NOT NULL
```
Comment (fedorloktev) [inline task2-sql-db-design/script.sql:32]: I am not sure 30 symbols limit is enough for account number

File: task2-sql-db-design/script.sql [hunk 3]
```diff
@@ -0,0 +1,76 @@
+CREATE TABLE item (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(50) NOT NULL,
+    description TEXT,
+    condition VARCHAR(50) NOT NULL,
+    category_id SERIAL,
+    seller_id SERIAL NOT NULL,
+    starting_price DECIMAL(9, 2),
+    auction_start TIMESTAMPTZ NOT NULL,
+    auction_end TIMESTAMPTZ
```
Comment (fedorloktev) [inline task2-sql-db-design/script.sql:10]: For discuss. I bet more correct business logic is having auction_end time as NOT NULL, is it? Somebody can see the end-time and make his bid when nobody already can't do a bid with upper price

File: task2-sql-db-design/script.sql [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    country VARCHAR(30) NOT NULL,
+    district VARCHAR(50) NOT NULL,
+    city VARCHAR(50) NOT NULL,
+    street VARCHAR(50) NOT NULL,
+    building VARCHAR(30) NOT NULL,
+    apartment VARCHAR(10) NOT NULL,
+    postal_code VARCHAR(30) NOT NULL
+);
+
+CREATE TABLE "group" (
```
Comment (fedorloktev) [inline task2-sql-db-design/script.sql:57]: Why do we have brackets here and on "user"table? Is it really needed?

**Comment by fedorloktev** [comment]
I offer to separate ITEM table for two tables (we spoke about it)

File: task2-sql-db-design/script.sql (hunk 4 above)
Comment (mfnsvrtm) [reply task2-sql-db-design/script.sql:57]: I believe there is not other way around it. `USER` and `GROUP` are both reserved keywords. You can find them both in this list: https://www.postgresql.org/docs/current/sql-keywords-appendix.html

**Pull request merged**